packages/yarnpkg-core/sources/types.ts

```diff
@@ -45,3 +45,23 @@ export interface Package extends Locator {
   // called at every keystroke)
   bin: Map<string, PortablePath>,
 }
+
+export enum PackageExtensionType {
+  Dependency = `Dependency`,
+  PeerDependency = `PeerDependency`,
+  PeerDependencyMeta = `PeerDependencyMeta`,
+}
```

Comment:
I had to move this enum over here (from `Configuration.ts`), as it produced a dependency cycle between `Configuration.ts` and `formatUtils.ts` that lead to runtime errors.